app/models/results/web_response_parser.rb

```diff
@@ -104,5 +105,9 @@ def inst_num(type, tree_parent)
         1
       end
     end
+
+    def invalid?(questioning_id)
+      FormItem.find(questioning_id).mission_id != @response.mission_id
```

Comment:
Nice! One issue with this though is that it will cause a DB query for every single ResponseNode, which is unnecessary.

As an alternative, you could have a method that returns a hash where the keys are IDs of all FormItems in the mission and the values are the same thing. (Try `FormItem.where(mission_id: @response.mission_id).pluck(:id).index_by(&:itself)`). This should be a memoized method.

Then you can simply do an _O(1)_ hash lookup to see if an incoming qing_id is in the mission.

```